include/seqan3/io/sequence/sequence_file_format_sam.hpp

```diff
@@ -0,0 +1,352 @@
+    //!\}
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "sam" },
+    };
+    //!\copydoc sequence_file_in_format_concept::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type,
+              typename seq_qual_type>
+    void read(stream_type                                         & stream,
+              sequence_file_in_options<seq_legal_alph_type> const & options,
+              seq_type                                            & sequence,
+              id_type                                             & id,
+              qual_type                                           & qualities,
+              seq_qual_type                                       & seq_qual)
+    {
+        static_assert(detail::decays_to_ignore_v<seq_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the sequence field, or the seq_qual field need to be set to std::ignore.");
+        static_assert(detail::decays_to_ignore_v<qual_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the qualities field, or the seq_qual field need to be set to std::ignore.");
+        auto stream_view = view::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                          decltype(std::istreambuf_iterator<char>{})>
+                            {std::istreambuf_iterator<char>{stream},
+                             std::istreambuf_iterator<char>{}};
+        auto stream_it = ranges::begin(stream_view);
+        if(*stream_it != '@') { //Ignore comment lines
+          // ID
+          read_id(stream_view, options, id);
+          // Sequence
+          if constexpr (!detail::decays_to_ignore_v<seq_type>) // sequence
+              read_seq(stream_view,  options, sequence);
+          else
+              read_seq(stream_view,  options, seq_qual);     // seq_qual (possibly std::ignore, too)
+          // Qualities
+         if constexpr (!detail::decays_to_ignore_v<qual_type>) // qualities
+              read_qual(stream_view,  qualities);
+          else
+              read_qual(stream_view, seq_qual);        // seq_qual (possibly std::ignore, too)
+              //TODO read_qual who returns seq_qual with correct output format
+          // make sure "buffer at end" implies "stream at end"
+          if ((std::istreambuf_iterator<char>{stream} == std::istreambuf_iterator<char>{}) &&
+              (!stream.eof()))
+          {
+              stream.get(); // triggers error in stream and sets eof
+          }
+        }
+        else{
+          while(*stream_it != '\n')
+          {++stream_it;}
```

Comment:
instead do `detail::consume(stream_view | view::take_line_or_throw);` This makes sure an end-of-line is actually reached (throws an exception otherwise).